src/features/MoviesPage/index.js

```diff
@@ -1,25 +1,53 @@
-import { Wrapper } from "./styled";
 import MovieCard from "./MovieCard";
 import Container from "../../common/Container";
 import Pagination from "../../common/Pagination";
 import Navigation from "../../common/Navigation";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchMovies, selectMovies, selectStatus } from "./slice";
+import { useEffect } from "react";
+import getMovies from "../../core/moviesAPI";
+import LoadingPage from "../../common/LoadingPage";
+import ErrorPage from "../../common/ErrorPage";
 
 const MoviesPage = () => {
-  return (
-    <>
-      <Navigation />
-      <Container title="Popular Movies">
-        <Wrapper>
-          <MovieCard />
-          <MovieCard />
-          <MovieCard />
-          <MovieCard />
-          <MovieCard />
-        </Wrapper>
-        <Pagination />
-      </Container>
-    </>
-  );
+  const movies = useSelector(selectMovies);
+  const status = useSelector(selectStatus);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchMovies(getMovies));
+  }, [dispatch]);
+
+  switch (status) {
+    case "loading":
+      return (
+        <>
+          <Navigation />
+          <LoadingPage />
+        </>
+      );
+
+    case "error":
+      return (
+        <>
+          <Navigation />
+          <ErrorPage />
+        </>
+      );
+
+    case "success":
+      return (
+        <>
+          <Navigation />
+          <Container title="Popular Movies">
+            <MovieCard
+              movies={movies}
+            />
```

Comment:
To trochę nie tak miało być, chodziło o to żeby tutaj był Wrapper, w którym generujesz dla każdego filmu MovieCard.
teraz przekazujesz wszystkie filmy do MovieCard, i wychodzi na to że MovieCard to 20 MovieCards 😅 Cała logika powinna być na poziomie MoviesPage, a MovieCard dostaje jedynie dane jednego filmu i je pokazuje. 😁